### DISCUSSION TITLE
Looking for a 120MM fan for my enclosure

### DISCUSSION THREAD

**Discussion by chadhobson** [tagyoureit/nodejs-poolController#796] [Nixie] [answered]
I've got a 24VDC PSU, 110VAC, and a Pi4 in the box. I've got the Altelix case with two 120MM openings on the door. Not sure I want one fan blowing out on top or if I want that plus one blowing in on the bottom. The Pi4 is right in front of the bottom fan so might be nice to run two fans.

At any rate, ideally they would only kick on when the Pi temp hit a certain level but I'm not sure how much is possible in regards to using the GPIO pins (assuming I use a 5V fan) or if others have come up with something better?

Thoughts?

**Comment by rstrouse** [comment, marked as answer]
You really do not need 2 fans.  The idea is to remove hot air from the enclosure and new cooler air will be drawn in from the bottom.  You can use the CPU temp device in REM to kick off a relay to engage even a 24vdc fan.

**Reply by chadhobson** [reply]
Excellent. I was hoping to go the 24V route for PSU simplicity. Thanks @rstrouse!

**Comment by chadhobson** [comment]
So much for simple. Got a 24VDC fan in today. Hooked it up straight to the 24VDC PSU in the enclosure and it works fine. One step closer.

Then I hook it up to one of the relays in the Sequent hat but when I turn the relay on in REM, I hear the click and see a red light come on the board but... nothing happens with the fan. Can't tell if I'm wired in the NO or NC side so I switch them but still nothing. 

I'm wondering if I'm missing something obvious again because again, two wires is about as simple as it gets. 

Thoughts?

P.S. Also got some more 10K thermistors in today and already added to the system for air temp.

**Reply by chadhobson** [reply]
Hooked up my multimeter to it and there's definitely no power coming through the relay. When I turn it on, it goes negative for a millisecond then back to zero. Wonder if i got a bad hat?

**Reply by chadhobson** [reply]
Got it now. All I had was some 12 gauge THHN and these connectors are way too small for it. Had the two wires touching  sticking out of the connector and didn't notice. Got some proper 18 gauge and all is well now. Disregard.

**Reply by rstrouse** [reply]
I was going to say it is most likely because you are using the wrong hookup wire in the connectors.  Just kidding but it would have been eerie as hell.  A box of multi-colored hookup wire spools is a good idea as well as a Spooky for low voltage connections.

**Reply by chadhobson** [reply]
Ha! Its always something. While I have you, whats the missing piece to turning the fan off and on based on Pi temp? I added the PI's temp as a generic device and then headed over to the 8 hat config to add a trigger but not really seeing anything that feels right.

![Screenshot 2023-06-22 at 7 29 30 PM](https://github.com/tagyoureit/nodejs-poolController/assets/1839161/9bee31cb-d7b9-430a-99b9-ab9b51dca23d)

**Reply by rstrouse** [reply]
Create a feed from a generic cpu temp device.

**Reply by chadhobson** [reply]
Finally getting back around to this after successfully adding a Watchdog to my enclosure. Yay me.

So if I go to my CPU Temp generic device and click on Feeds, I see this:

![Screenshot 2023-06-26 at 4 09 39 PM](https://github.com/tagyoureit/nodejs-poolController/assets/1839161/899ffa9e-0d35-45a0-8906-a15704483a49)

If I select the relay hat as the device to send to, the only option in the dropdown is relayState which crashes REM if I select it.

```
Jun 26 16:03:53 nixie node[741]: /home/chadhobson/relayEquipmentManager/i2c-bus/I2c-Relay.ts:941
Jun 26 16:03:53 nixie node[741]:             let oldState = relay.state || false;
Jun 26 16:03:53 nixie node[741]:                                  ^
Jun 26 16:03:53 nixie node[741]: TypeError: Cannot read properties of undefined (reading 'state')
Jun 26 16:03:53 nixie node[741]:     at i2cRelayMulti.setRelayState (/home/chadhobson/relayEquipmentManager/i2c-bus/I2c-Relay.ts:941:34)
Jun 26 16:03:53 nixie node[741]:     at i2cRelayMulti.setValue (/home/chadhobson/relayEquipmentManager/i2c-bus/I2c-Relay.ts:934:22)
Jun 26 16:03:53 nixie node[741]:     at i2cBus.setDeviceValue (/home/chadhobson/relayEquipmentManager/i2c-bus/I2cBus.ts:359:51)
Jun 26 16:03:53 nixie node[741]:     at i2cController.setDeviceValue (/home/chadhobson/relayEquipmentManager/i2c-bus/I2cBus.ts:145:45)
Jun 26 16:03:53 nixie node[741]:     at InternalConnection.send (/home/chadhobson/relayEquipmentManager/connections/Bindings.ts:147:31)
Jun 26 16:03:53 nixie node[741]:     at Feed.send (/home/chadhobson/relayEquipmentManager/boards/Controller.ts:978:35)
Jun 26 16:03:53 nixie node[741]:     at CPUTempDevice.emitFeeds (/home/chadhobson/relayEquipmentManager/generic/genericDevices.ts:233:37)
Jun 26 16:03:53 nixie node[741]:     at CPUTempDevice.takeReadings (/home/chadhobson/relayEquipmentManager/generic/SystemDevices.ts:115:18)
Jun 26 16:03:53 nixie node[741]:     at runMicrotasks (<anonymous>)
Jun 26 16:03:53 nixie node[741]:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
Jun 26 16:03:53 nixie node[741]:     at /home/chadhobson/relayEquipmentManager/generic/SystemDevices.ts:126:21
Jun 26 16:03:53 nixie systemd[1]: nixie-relay.service: Main process exited, code=exited, status=1/FAILURE
Jun 26 16:03:53 nixie systemd[1]: nixie-relay.service: Failed with result 'exit-code'.
Jun 26 16:03:53 nixie systemd[1]: nixie-relay.service: Consumed 19.879s CPU time.
Jun 26 16:04:08 nixie systemd[1]: nixie-relay.service: Scheduled restart job, restart counter is at 1.
```

Feel like I've looked at this from every angle and I'm not seeing it.

**Reply by chadhobson** [reply]
I think I've got it now. [This post](https://github.com/tagyoureit/nodejs-poolController/discussions/405) got me over the hump. I've got the fan kicking on if the CPU temp is greater than 120 F. I'll tweak that number once its installed on the pad.
